.github/workflows/tests-npm.yml

```diff
@@ -72,4 +89,94 @@ jobs:
         run: ${{ inputs.pre_test_command }}
         if: inputs.pre_test_command != ''
       - name: Run tests
-        run: npm run ${{ matrix.command }}
+        run: npx c8 npm run ${{ matrix.command }}
+      - name: Set artifact name
+        id: artifact-name
+        run: echo "name=coverage-$(echo "${{ matrix.command }}" | sed 's/[:\";<>|*?\/\\ \r\n]/-/g')" >> $GITHUB_OUTPUT
+        if: inputs.coverage
+      - name: Upload coverage JSON summary
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ steps.artifact-name.outputs.name }}
+          path: ${{ steps.coverage-dir.outputs.path }}
+        if: inputs.coverage
+
+  coverage:
+    name: Coverage
+    runs-on: ubuntu-latest
+    needs: tests
+    if: inputs.coverage
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: ${{ inputs.node_version }}
+
+      - name: Setup c8 config
+        run: |
+          if [ -n "${{ inputs.coverage_config }}" ]; then
+            echo "Copying repo config '${{ inputs.coverage_config }}' to .c8rc.workflow.json"
+            cp ${{ inputs.coverage_config }} .c8rc.workflow.json
+          else
+            echo "Using default .c8rc.workflow.json"
+            cat > .c8rc.workflow.json << 'EOF'
+          {
+            "temp-directory": "coverage/tmp",
+            "reporter": ["text", "html"],
+            "lines": 50,
+            "branches": 0,
+            "functions": 0,
+            "statements": 0,
+            "exclude": [
+              "coverage/**",
+              "packages/*/test{,s}/**",
+              "**/*.d.ts",
+              "test{,s}/**",
+              "test{,-*}.js",
+              "**/*{.,-}test.js",
+              "**/__tests__/**",
+              "**/node_modules/**",
+              "**/babel.config.js",
+              "**/*.tsx"
+            ]
```

Comment:
should we exclude `**/routes.ts` by default as well, since we use `tsoa` so much. Risk is a non auto generated file called routes.ts is ignored